File: works/tmp/1476.java
import java.sql.Array;
import java.util.ArrayList;
import java.util.Collections;
import java.util.Scanner;

public class Main
{
    public static void main(String[] args)
    {
        Scanner sc = new Scanner(System.in);
        int E, S, M;
        E = sc.nextInt(); S = sc.nextInt(); M = sc.nextInt();
        int cnt = 1;

        int e = 1, s = 1, m = 1;
        while(true)
        {
            if(e == E && s == S && m == M) break;
            cnt++;

            if(e < 15) e++;
            else e = 1;

            if(s < 28) s++;
            else s = 1;

            if(m < 19) m++;
            else m = 1;
        }
        System.out.print(cnt);
    }
}

File: works/tmp/1476.kt
import java.util.*
import kotlin.collections.ArrayList

fun main()
{
    val sc = Scanner(System.`in`)
    var E = sc.nextInt()
    var S = sc.nextInt()
    var M = sc.nextInt()

    var cnt = 1
    var e = 1; var s = 1; var m = 1
    while(true)
    {
        if(e == E && s == S && m == M) break;
        cnt++;

        if(e < 15) e++;
        else e = 1;

        if(s < 28) s++;
        else s = 1;

        if(m < 19) m++;
        else m = 1;
    }
    print(cnt)
}